src/lib/templating.ts

```diff
@@ -0,0 +1,160 @@
+/**
+ * This module handles template interpolation.
+ */
+
+import { StepMatcher } from '@/lib/matcher';
+import * as S from '@/lib/steps';
+
+export type ScopeContext = {
+  [varname: string]: any;
+};
+
+export type Interpolator = (template: string, context: ScopeContext) => any;
+type StepInterpolator = (step: S.PipelineStep) => S.PipelineStep;
+type ConditionType = S.FilterSimpleCondition | S.FilterComboAnd | S.FilterComboOr;
+
+function _interpolate(interpolator: Interpolator, value: any, context: ScopeContext) {
+  if (typeof value === 'string') {
+    return interpolator(value, context);
+  }
+  return value;
+}
+
+function interpolateFilterCondition(
+  condition: ConditionType,
+  interpolator: Interpolator,
+  context: ScopeContext,
+): ConditionType {
+  if (S.isFilterComboAnd(condition)) {
+    return {
+      and: condition.and.map(cond => interpolateFilterCondition(cond, interpolator, context)),
+    };
+  } else if (S.isFilterComboOr(condition)) {
+    return {
+      or: condition.or.map(cond => interpolateFilterCondition(cond, interpolator, context)),
+    };
+  } else {
+    switch (condition.operator) {
+      case 'eq':
+      case 'ne':
+      case 'gt':
+      case 'ge':
+      case 'lt':
+      case 'le':
+        return { ...condition, value: _interpolate(interpolator, condition.value, context) };
+      case 'in':
+      case 'nin':
+      default:
+        // only for typescript to be happy and see we always have a return value
+        return {
+          ...condition,
+          value: condition.value.map(v => _interpolate(interpolator, v, context)),
+        };
+    }
+  }
+}
+
+/**
+ * The `Interpolator` class provides an "step interpolator" implementation
+ * for each possible step in a pipeline.
+ * A step interpolator is a function that takes a step in parameter and returns
+ * the "interpolated" version of a step, given a specific interpolate function (e.g.
+ * `_.template`).
+ */
+export class PipelineInterpolator implements StepMatcher<S.PipelineStep> {
```

Comment:
Ok with the list of steps and parameters supporting templating for a first version! Thanks